keras/layers/recurrent.py

```diff
@@ -66,6 +67,75 @@ def output(self, train):
         return outputs[-1]
 
 
+
+class SimpleBRNN(Layer):
+    '''
+        Fully connected Bi-directional RNN where:
+            Output at time=t is fed back to input for time=t+1 in a forward pass
+            Output at time=t is fed back to input for time=t-1 in a backward pass
+    '''
+    def __init__(self, input_dim, output_dim,
+        init='uniform', inner_init='orthogonal', activation='sigmoid', weights=None,
+        truncate_gradient=-1,  return_sequences=False):
+        self.init = initializations.get(init)
+        self.inner_init = initializations.get(inner_init)
+        self.input_dim = input_dim
+        self.input_dim = input_dim
+        self.output_dim = output_dim
+        self.truncate_gradient = truncate_gradient
+        self.activation = activations.get(activation)
+        self.return_sequences = return_sequences
+        self.input = T.tensor3()
+
+        self.W_if = self.init((self.input_dim, self.output_dim/2))    # Input -> Forward
+        self.W_ff = self.init((self.output_dim/2, self.output_dim/2))   # Forward tm1 -> Forward t
+        self.W_ib = self.init((self.input_dim, self.output_dim/2))    # Input -> Backward
+        self.W_bb = self.init((self.output_dim/2, self.output_dim/2))   # Backward t -> Backward tm1
+        self.b_f = shared_zeros((self.output_dim/2))
+        self.b_b = shared_zeros((self.output_dim/2))
```

Comment:
This divides the input into halves of size output_dim/2 that are then concatenated. The size of the output might not be output_dim anymore (eg: python(13/2) = 6; 6+6=12).